app/grandchallenge/algorithms/models.py

```diff
@@ -967,13 +971,13 @@ def retrieve_existing_civs(*, civ_data):
         return existing_civs
 
     def get_jobs_with_same_inputs(
-        self, *, inputs, algorithm_image, algorithm_model
+        self, *, inputs, interface, algorithm_image, algorithm_model
     ):
         existing_civs = self.retrieve_existing_civs(civ_data=inputs)
         unique_kwargs = {
             "algorithm_image": algorithm_image,
         }
-        input_interface_count = algorithm_image.algorithm.inputs.count()
+        input_interface_count = interface.inputs.count()
```

Comment:
I'm unsure about this one. I'm wondering now if it would be enough to count the provided inputs and first filter on jobs with that number of inputs, and then do the matching inputs annotation. Then we'd be checking for jobs with exactly the inputs provided, right?

Reply:
That check is already being done on L990 with `.filter(inputs_match_count=input_interface_count)` right?

Reply:
On that line we're currently checking if the number of matching inputs match those in the interface that we pass to this function. But I'm wondering if we need to pass the interface to this function at all. All we want to know if there are any jobs with the provided inputs, so it should be enough to check whether the `inputs_match_count=len(inputs)`, right?

Reply:
What if you have inputs that are subsets of another e.g. `{A, C}` and `{A, B}`?

Reply:
That case is covered. This function would not return jobs with partially overlapping inputs. I have also added a test for that. 

I have opted to not pass the interface to this function as it does not seem to be necessary to me.